bootloader/bootloader.go

```diff
@@ -352,8 +352,17 @@ func ForceError(err error) {
 	forcedError = err
 }
 
-func extractKernelAssetsToBootDir(dstDir string, snapf snap.Container, assets []string) error {
-	// now do the kernel specific bits
+// extractKernelAssetsToBootDir will extract the given assets into the
+// given assetsDir. If the target directory already exists it will do
+// nothing.
+func extractKernelAssetsToBootDir(assetsDir string, snapf snap.Container, assets []string) error {
+	// Final dir already in place: nothing more to do.
```

Comment:
personally I think it's safer to check that the contents of the directory has the expected assets we need rather than assume that the if the directory exists that it has all the assets, and I don't think that checking this will take much more time, we already will be speeding things up by copying just once instead of twice. I don't have a clear idea of how we could get into that situation given how we extract the assets now with a rename instead of copying assets individually, but I still think we could be more robust here IMHO. If it's thought to be overkill though I don't feel that strongly about it

Reply:
the description is confusing on this point. From the description it sounded like we would unpack anyway and check for the dir afterward. Anyway I think I agree with @anonymouse64 that checking for the content seems more robust, match file  list, sizes, and maybe even some light checks about the start and end of files

Reply:
I was thinking about this too, it seems like quite a bit of code (and tests) for something we should never run into with the structure in place. But I can add if of course if both you and Samuele prefer it this way.

Reply:
so we discussed this a bit more and a reason to want to check the content is the case of partially removed directories

Reply:
But directory removal can also happen atomically:
1. You rename the `assetsDir` to `assetsDir + ".removed"`
2. You remove `assetsDir + ".removed"`

Then, you can be sure that if `assetsDir` already exists, it's complete. Or did I misunderstand the problem?

Reply:
Yeah, remove also needs to rename first and then remove to avoid this issue and deal with two dirs

Reply:
Yep. And that means that you don't have to check the directory contents, right?